src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/ThingRegistry.java

```diff
@@ -116,32 +116,27 @@ private Thing storeThing(Thing thing) {
     /**
      * Returns whether the Thing is associated to the given IoT Certificate.
      * Returns valid locally registered result when IoT Core cannot be reached.
-     * TODO: add a separate refreshable caching layer for offline auth
      *
      * @param thing IoT Thing
      * @param certificate IoT Certificate
      * @return whether thing is attached to the certificate
      * @throws CloudServiceInteractionException when thing <-> certificate association cannot be verified
      */
     public boolean isThingAttachedToCertificate(Thing thing, Certificate certificate) {
-        try {
-            if (iotAuthClient.isThingAttachedToCertificate(thing, certificate)) {
-                thing.attachCertificate(certificate.getCertificateId());
-                updateThing(thing);
-                return true;
-            } else {
-                thing.detachCertificate(certificate.getCertificateId());
-                updateThing(thing);
-            }
-        } catch (CloudServiceInteractionException e) {
-            // TODO: also check the instant when this cert was attached this thing
-            //  and validate against configured offline TTL
-            if (thing.isCertificateAttached(certificate.getCertificateId())) {
-                return true;
-            }
-            throw e;
+        // If we have a local cache hit, then return true. Else, go to cloud
+        if (thing.isCertificateAttached(certificate.getCertificateId())) {
+            return true;
+        }
+
+        if (iotAuthClient.isThingAttachedToCertificate(thing, certificate)) {
+            thing.attachCertificate(certificate.getCertificateId());
```

Comment:
is this a no-op when the cert is already "attached"?

Reply:
Well, it won't get here if it's already attached so it's kind of moot. But, ignoring that, re-attaching will update the timestamp which effectively extends the TTL.

None of that is here yet. I'm expecting there to even be a little more factoring, but this was the minimal change I could make to get this working.